src/scipp/html/table.py

```diff
@@ -0,0 +1,230 @@
+        out += f'<th colspan="{len(ds.coords)}"></th>'
+    for name, da in sorted(ds.items()):
+        ncols = 1 + len(da.masks) + len(da.attrs)
+        out += f'<th {CENTER} colspan="{ncols}">{name}</th>'
+    out += '</tr>'
+    return out
+
+
+def _make_sections_header(ds: Dataset) -> str:
+    out = '<tr>'
+    if ds.coords:
+        out += f'<th {CENTER} colspan="{len(ds.coords)}">Coordinates</th>'
+    for _, da in sorted(ds.items()):
+        out += f'<th {CENTER_BORDER}>Data</th>'
+        if da.masks:
+            out += f'<th {CENTER} colspan="{len(da.masks)}">Masks</th>'
+        if da.attrs:
+            out += f'<th {CENTER} colspan="{len(da.attrs)}">Attributes</th>'
+    out += '</tr>'
+    return out
+
+
+def _to_html_table(header: str, body: List[list]) -> str:
+    out = '<table>' + header
+    ncols = len(body)
+    nrows = len(body[0])
+    for i in range(nrows):
+        out += '<tr>' + ''.join([body[j][i] for j in range(ncols)]) + '</tr>'
+    out += '</table>'
+    return out
+
+
+def _find_bin_edges(ds: Dataset) -> bool:
+    for key in ds.coords:
+        if ds.coords.is_edges(key):
+            return True
+    for da in ds.values():
+        for key in da.attrs:
+            if da.attrs.is_edges(key):
+                return True
+    return False
+
+
+def _strip_scalars_and_broadcast_masks(ds: Dataset) -> Dataset:
+    out = Dataset()
+    for key, da in ds.items():
+        if da.ndim == 1:
+            out[key] = DataArray(data=da.data,
+                                 coords={
+                                     key: var
+                                     for key, var in da.coords.items()
+                                     if var.dims == da.data.dims
+                                 },
+                                 attrs={
+                                     key: var
+                                     for key, var in da.attrs.items()
+                                     if var.dims == da.data.dims
+                                 },
+                                 masks={
+                                     key: var.broadcast(sizes=da.sizes)
+                                     for key, var in da.masks.items()
+                                 })
+    return out
+
+
+def _to_dataset(obj: Union[VariableLike, dict]) -> Dataset:
+    if isinstance(obj, DataArray):
+        return Dataset({obj.name: obj})
+    if isinstance(obj, Variable):
+        return Dataset(data={"": obj})
+    if isinstance(obj, dict):
+        return Dataset(obj)
+    return obj
+
+
+def table(obj: Union[VariableLike, dict], max_rows: Optional[int] = 20):
+    """Create a html table from the contents of the supplied object.
+
+    Possible inputs are:
+     - Variable
+     - DataArray
+     - Dataset
+     - dict of Variable
+     - dict of DataArray
+
+    Inputs must be one-dimensional. Zero-dimensional data members, attributes and
+    coordinates are stripped. Zero-dimensional masks are broadcast.
```

Comment:
Why treat masks and coords/attrs differently?

Reply:
So would just just broadcast everything?

Reply:
No, this would be noisy when there are many scalar attrs. But why did you broadcast the mask? It means that either all or no elements are masked. Could this be represented more neatly?

Reply:
Because I think the mask information is still important. If there is a 0D mask, it will apply to everything. Just like when you plot things, you need to broadcast the masks.

Reply:
It might not be entirely clear that the mask affects all elements when the data is cut off in the table. But I don't see a particularly good way of representing that. Maybe this?
```
Data Masks
[m]   mask
0.1   1
0.2   ...
```

Also, just realised that masks are represented numerically. I think showing them as 'True' / 'False' would be better. (In the off chance that as mask has different values, casting to `bool` should still be fine here?)

Reply:
I thought that having as numbers might be nice if someone wants to copy/paste some values into a csv or something.

Reply:
I think `True`/`False` is preferable? Not everyone knows how those map to integers?